main.py

```diff
@@ -895,6 +895,21 @@ def patch_record_in_place(fn, record, subdir):
     # track_features #
     ##################
 
+    # Replace versioned Linux sysroot features with non-versioned ones so they properly weight newer and older sysroot
+    # versions. Similarly, add features to sysroots that do not have at least one feature.
+    if subdir == "noarch" and name.startswith("sysroot_linux") and VersionOrder(version) < VersionOrder("2.28"):
+        new_features = set()  # Features should be unique.
+        if "track_features" in record:
+            for feature in record["track_features"].split():
+                if re.search(r'_[0-9]+(\.[0-9]+)+$', feature):
+                    new_features.add(feature.rsplit("_", 1)[0])  # Strip the version from the feature if it has one.
```

Comment:
You're doing the `_` "split" twice.  Any reason that might be clearer if I knew what the spec of a "feature" might be?

Personally, I might have written:

```suggestion
                mo = re.search(r'_([0-9]+(\.[0-9]+)+)$', feature)
                if mo:
                    new_features.add(mo.group(1))
```